parquet/src/geospatial/bounding_box.rs

```diff
@@ -0,0 +1,429 @@
+    zmax: Option<f64>,
+    /// Minimum M coordinate (measured value), if present
+    mmin: Option<f64>,
+    /// Maximum M coordinate (measured value), if present
+    mmax: Option<f64>,
+}
+
+impl BoundingBox {
+    /// Creates a new bounding box with the specified coordinates.
+    pub fn new(xmin: f64, xmax: f64, ymin: f64, ymax: f64) -> Self {
+        Self {
+            xmin,
+            xmax,
+            ymin,
+            ymax,
+            zmin: None,
+            zmax: None,
+            mmin: None,
+            mmax: None,
+        }
+    }
+
+    /// Updates the bounding box with specified X-coordinate range.
+    pub fn with_xrange(mut self, xmin: f64, xmax: f64) -> Self {
+        self.xmin = xmin;
+        self.xmax = xmax;
+        self
+    }
+
+    /// Updates the bounding box with specified Y-coordinate range.
+    pub fn with_yrange(mut self, ymin: f64, ymax: f64) -> Self {
+        self.ymin = ymin;
+        self.ymax = ymax;
+        self
+    }
+
+    /// Creates a new bounding box with the specified Z-coordinate range.
+    pub fn with_zrange(mut self, zmin: f64, zmax: f64) -> Self {
+        self.zmin = Some(zmin);
+        self.zmax = Some(zmax);
+        self
+    }
+
+    /// Creates a new bounding box with the specified M-coordinate range.
+    pub fn with_mrange(mut self, mmin: f64, mmax: f64) -> Self {
+        self.mmin = Some(mmin);
+        self.mmax = Some(mmax);
+        self
+    }
+
+    /// Returns the minimum x-coordinate.
+    pub fn get_xmin(&self) -> f64 {
+        self.xmin
+    }
+
+    /// Returns the maximum x-coordinate.
+    pub fn get_xmax(&self) -> f64 {
+        self.xmax
+    }
+
+    /// Returns the minimum y-coordinate.
+    pub fn get_ymin(&self) -> f64 {
+        self.ymin
+    }
+
+    /// Returns the maximum y-coordinate.
+    pub fn get_ymax(&self) -> f64 {
+        self.ymax
+    }
+
+    /// Returns the minimum z-coordinate, if present.
+    pub fn get_zmin(&self) -> Option<f64> {
+        self.zmin
+    }
+
+    /// Returns the maximum z-coordinate, if present.
+    pub fn get_zmax(&self) -> Option<f64> {
+        self.zmax
+    }
+
+    /// Returns the minimum m-value (measure), if present.
+    pub fn get_mmin(&self) -> Option<f64> {
+        self.mmin
+    }
+
+    /// Returns the maximum m-value (measure), if present.
+    pub fn get_mmax(&self) -> Option<f64> {
+        self.mmax
+    }
+
+    /// Returns `true` if both zmin and zmax are present.
+    pub fn is_z_valid(&self) -> bool {
+        self.zmin.is_some() && self.zmax.is_some()
+    }
+
+    /// Returns `true` if both mmin and mmax are present.
+    pub fn is_m_valid(&self) -> bool {
+        self.mmin.is_some() && self.mmax.is_some()
+    }
+}
+
+impl From<BoundingBox> for parquet::BoundingBox {
+    /// Converts our internal `BoundingBox` to the Thrift-generated format.
+    fn from(b: BoundingBox) -> parquet::BoundingBox {
+        parquet::BoundingBox {
+            xmin: b.get_xmin().into(),
```

Comment:
Tiny nit: since we have access to the private attributes here, it might be slightly more readable to use those, but up to you:

```suggestion
            xmin: b.xmin.into(),
```